backend/app/api/api_v1/endpoints/job.py

```diff
@@ -0,0 +1,111 @@
+from typing import Any, List
+
+from fastapi import APIRouter, Body, Depends, HTTPException
+from sqlalchemy.orm import Session
+
+from app import crud, schemas
+from app.api import deps
+from app.core.config import settings
+
+router = APIRouter()
+
+@router.get("/all", response_model=List[schemas.Job])
+def get_all_job(
+    db: Session = Depends(deps.get_db),
+    skip: int = 0,
+) -> Any:
+    """
+    Retrieve all jobs.
+    """
+    jobs = crud.job.get_multi(db, skip=skip)
+    if not jobs:
+        raise HTTPException(
+            status_code=404,
+            detail="Jobs not found",
+        )
+    return jobs
+
+@router.get("/{job_id}", response_model=schemas.Job)
+def get_job_by_id(
+    job_id: int,
+    db: Session = Depends(deps.get_db),
+) -> Any:
+    """
+    Get a specific job by id.
+    """
+    job = crud.job.get(db, job_id=job_id)
+    if not job:
+        raise HTTPException(
+            status_code=404,
+            detail="Job not found",
+        )
+    return job
+
+@router.post("", response_model=schemas.Job)
+def create_job(
+    *,
+    db: Session = Depends(deps.get_db),
+    job_name: str = Body(...),
+    job_desc: str = Body(...),
+    is_active: bool
+) -> Any:
+    """
+    Create new job.
+    """
+    if not settings.USERS_OPEN_REGISTRATION:
+        raise HTTPException(
+            status_code=403,
+            detail="Open staff registration is forbidden on this server",
+        )
+    job_in = schemas.JobCreate(
+        job_name=job_name,
+        job_desc=job_desc,
+        is_active=is_active,
```

Comment:
This implies that all three attributes are compulsory for this endpoint. Is this intentional? Also, do we want to "automate" the `is_active` value? i.e. if creating a job, `is_active` is automatically set as `True` (just a suggestion)

Reply:
done, now creating job only needs name and desc as is_active will be set to True by default